common/common.go

```diff
@@ -38,19 +38,22 @@ type BaseKvdb struct {
 	lock sync.Mutex
 }
 
+// SetFatalCb sets fatal func callback.
 func (b *BaseKvdb) SetFatalCb(f kvdb.FatalErrorCB) {
 	b.lock.Lock()
 	defer b.lock.Unlock()
 	b.FatalCb = f
 }
 
+// SetLockTimeout sets lock timeout.
```

Comment:
If the lock is held past this duration, then a configured fatal callback is called.